kubemarine/plugins/calico.py

```diff
@@ -228,45 +224,18 @@ def enrich_daemonset_calico_node_container_env(self, container_pos: int, contain
                 'CALICO_IPV6POOL_CIDR', 'IP6', 'IP6_AUTODETECTION_METHOD',
                 'CALICO_IPV6POOL_IPIP', 'CALICO_IPV6POOL_VXLAN'
             ])
-        if utils.true_or_false(self.inventory['plugins']['calico']['typha']['enabled']) == "false":
+        if not self.is_typha_enabled():
             env_delete.append('FELIX_TYPHAK8SSERVICENAME')
 
-        for name in env_delete:
-            for i, e in enumerate(container['env']):
-                if e['name'] == name:
-                    del container['env'][i]
-                    self.log.verbose(f"The {name!r} env variable has been removed from "
-                                    f"'spec.template.spec.containers.[{container_pos}].env' in the {key}")
-                    break
-
-        env_update: Dict[str, dict] = {}
-        for name, value in self.inventory['plugins']['calico']['env'].items():
-            if name in env_delete:
-                continue
-            if type(value) is str:
-                env_update[name] = {'value': value}
-            elif type(value) is dict:
-                env_update[name] = {'valueFrom': value}
-            self.log.verbose(f"The {key} has been patched in "
-                            f"'spec.template.spec.containers.[{container_pos}].env.{name}' with '{value}'")
-
-        for env in container['env']:
-            name = env['name']
-            if name not in env_update:
-                continue
-
-            value = env_update.pop(name)
-            keys = list(env.keys())
-            for key in keys:
-                if key != 'name' and key not in value:
-                    del env[key]
-
-            env.update(value)
-
-        for name, env in env_update.items():
-            new_env = {'name' : name}
-            new_env.update(env)
-            container['env'].append(new_env)
+        env_ensure: Dict[str, str] = {
+            # If metrics ports are ever configurable,
+            # it makes sense to make it configurable for typha and kube-controllers as well.
+            # Metrics services should also be patched.
+            'FELIX_PROMETHEUSMETRICSPORT': '9091'
+        }
```

Comment:
Previously it was possible to redefine this env in the inventory. But we do not take into account that services should also be patched correspondingly, and thus we have no full support of metrics port configuration.

Support of custom ports was not a goal of this PR, so I just restricted customization of this env. Although it is easy to customize port for `calico-node`, it is less easy to do that for `calico-typha` (need new inventory section), and definitely not easy to do that for `calico-kube-controllers`.

I would still like to know if I should do additionally in this PR:
1. Support customization of ports for all metrics
2. Support customization only of `calico-node` and `calico-typha` as they listen the host port.